### PR TITLE
Add another function to sort ims as per the pattern order given

### PR DESCRIPTION
**Pull request by SnowNooDLe**
# Sort IMs

IMs after implement this sorting function.

Screenshot from 2020-09-11 11-31-42

### CODE REVIEW

File: qcore/im.py [hunk 1]
```diff
@@ -88,3 +90,58 @@ def order_im_cols_df(df, pattern_order=default_pattern_order):
         [adj_cols.append(col) for col in orig_cols if col not in adj_cols]
 
     return df[adj_cols]
+
+
+def order_ims(unsorted_ims, pattern_order=default_pattern_order):
```
Comment (jasonmotha) [inline qcore/im.py:95]: Might be above your pay grade. But merge this function and the one above to limit the duplication of the code.

**Review by jasonmotha — CHANGES_REQUESTED** [review]

File: qcore/im.py [hunk 2]
```diff
@@ -88,3 +90,58 @@ def order_im_cols_df(df, pattern_order=default_pattern_order):
...
+                for col in cur_cols:
+                    values.extend(
+                        (
+                            list(
+                                float(col.split("_")[value_ix].replace("p", "."))
+                                for value_ix in float_cols
+                            )
+                        )
+                    )
+
```
Comment (sungeunbae) [inline qcore/im.py:132]: This part looks redundant - could be merged with the for loop above?

File: qcore/im.py [hunk 3]
```diff
@@ -27,85 +27,40 @@ def order_im_cols_file(filename):
     """For a full description see function order_im_cols_df"""
     df = pd.read_csv(filename)
 
-    return order_im_cols_df(df)
+    return order_im_cols_df(df.columns)
```
Comment (jasonmotha) [inline qcore/im.py:30]: Should pass the whole df not just the columns

**Review by jasonmotha — CHANGES_REQUESTED** [review]

**Review by jasonmotha — DISMISSED** [review]

File: qcore/im.py [hunk 4]
```diff
@@ -30,20 +30,30 @@ def order_im_cols_file(filename):
...
-def order_im_cols_df(df, pattern_order=default_pattern_order):
-    """Orders the columns in the dataframe as per the pattern order given.
+def order_im_cols_df(df, pattern_order=DEFAULT_PATTERN_ORDER):
+    """
+    Orders the columns in the dataframe as per the pattern order given.
+
+    All columns that don't match a pattern are just appended to the end in the
+    original order.
+    """
+
```
Comment (claudio525) [inline qcore/im.py:40]: empty line?

File: qcore/im.py [hunk 5]
```diff
@@ -30,20 +30,30 @@ def order_im_cols_file(filename):
...
     If there are several columns matching a pattern, and the column contains a
     number seperated by a '_', such as pSA_0.5, then those columns are sorted
     lowest to highest based on the number. The number has to be in the same
     position for all column names of a pattern.
-
-    All columns that don't match a pattern are just appended to the end in the
-    original order.
     """
-    orig_cols = df.columns
+    orig_cols = unsorted_ims
```
Comment (claudio525) [inline qcore/im.py:55]: Just rename orig_cols to unsorted_ims, then you can remove this line?

File: qcore/im.py (hunk 4 above)
Comment (SnowNooDLe) [reply qcore/im.py:40]: Ow, I thought I did apply Black in PyCharm, will check again on Monday as I turn off my PCs these days, cannot access remotely hahaha

**Review by claudio525 — APPROVED** [review]

**Review by jasonmotha — APPROVED** [review]

**Pull request merged**